From bb32afffa102742fe45b1ccf951486dcf01e7071 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:55:04 -0400
Subject: [fs] ext4: silence sparse check warning for function ext4_trim_extent

Message-id: <1414058125-4183-151-git-send-email-lczerner@redhat.com>
Patchwork-id: 98258
O-Subject: [RHEL 7.1 PATCH v2 150/171] ext4: silence sparse check warning for function ext4_trim_extent
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: jon ernst <jonernst07@gmail.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit e2cbd587418251bb73c4c1e8e2c7c1816d7a98d9

 This fixes the following sparse warning:

      CHECK   fs/ext4/mballoc.c
    fs/ext4/mballoc.c:5019:9: warning: context imbalance in
    'ext4_trim_extent' - unexpected unlock

 Signed-off-by: "Jon Ernst" <jonernst07@gmail.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/mballoc.c b/fs/ext4/mballoc.c
index 73ccbb3..c8238a2 100644
--- a/fs/ext4/mballoc.c
+++ b/fs/ext4/mballoc.c
@@ -5016,6 +5016,8 @@ error_return:
  */
 static int ext4_trim_extent(struct super_block *sb, int start, int count,
         ext4_group_t group, struct ext4_buddy *e4b)
+__releases(bitlock)
+__acquires(bitlock)
 {
  struct ext4_free_extent ex;
  int ret = 0;
-- 
1.7.1